lib/iris/common/metadata.py

```diff
@@ -54,6 +57,55 @@
 logger = get_logger(__name__, fmt="[%(cls)s.%(funcName)s]")
 
 
+_num2date_original = cf_units.Unit.num2date
+
+
+def _num2date_to_nearest_second(
+    self,
+    time_value,
+    only_use_cftime_datetimes=True,
+    only_use_python_datetimes=False,
+):
+    # Used to monkey-patch the cf_units.Unit.num2date method to round to the
+    #  nearest second, which was the legacy behaviour. This is under a FUTURE
+    #  flag - users will need to adapt to microsecond precision eventually,
+    #  which may involve floating point issues.
+    def _round(date):
+        if date.microsecond == 0:
+            return date
+        elif date.microsecond < 500000:
+            return date - timedelta(microseconds=date.microsecond)
+        else:
+            return (
+                date + timedelta(seconds=1) - timedelta(microseconds=date.microsecond)
+            )
+
+    result = _num2date_original(
+        self, time_value, only_use_cftime_datetimes, only_use_python_datetimes
+    )
+    if FUTURE.date_microseconds is False:
+        message = (
+            "You are using legacy date precision for Iris units - max "
+            "precision is seconds. In future, Iris will use microsecond "
+            "precision - from cf-units version 3.3 - which may affect core "
```

Comment:
I could instead say "available since cf-units version 3.3" if it were felt to be important. I'm uncertain how much it matters.

Reply:
that does sound clearer to me

Reply:
d673480cf8eb6da7e1c2fac983417e8658263a82